packages/supi/src/install/index.ts

```diff
@@ -403,20 +403,32 @@ export async function mutateModules (
       })
     }
 
+    function checkDependencyInPackage (dependencySet: Set<string>, inputs: string[]) {
+      let noneDependencyfound = true
+      for (const input of inputs) {
+        if (!dependencySet.has(input)) {
+          console.log(`No ${input} package found in dependencies of the project`)
+        } else {
+          noneDependencyfound = false
+        }
+      }
+      if (noneDependencyfound) {
+        throw new PnpmError('NO_PACKAGE_IN_DEPENDENCY', `No package found in dependencies of the project`)
+      }
+    }
+
     if (maybeOpts.strict) {
       const project = projectsToInstall[0]
-      let dependenciesList: string[] = []
+      let dependencySet: Set<string> = new Set()
       switch (project.mutation) {
         case 'installSome': {
           if (project.allowNew === false) {
             for (const p of projectsToInstall) {
-              dependenciesList = [...p.wantedDependencies.map(wantedDependency => wantedDependency.raw), ...dependenciesList]
-            }
-            for (const input of project.dependencySelectors) {
-              if (!dependenciesList.includes(input)) {
-                throw new PnpmError('NO_PACKAGE_IN_DEPENDENCY', `No ${input} package found in dependencies of the project`)
-              }
+              p.wantedDependencies.forEach(wantedDependency => {
+                dependencySet.add(wantedDependency.raw)
```

Comment:
use `wantedDependency.alias` instead of `.raw`

Reply:
alias is not working when dependencySelectors is something line `peer-2:workspace:*`

Reply:
ok, I'll look into that

Reply:
just checked, the alias is correct even when raw is `peer-2@workspace:*`.